ruby/lib/transformer/syskit/extensions/system_network_generator.rb

```diff
@@ -59,39 +70,77 @@ def validate_abstract_network
+                    SystemNetworkGeneratorExtension
+                        .validate_frame_selection_consistency_through_inputs(
+                            task, error_handler
+                        )
                 end
             end
         end
 
-        def self.validate_frame_selection_consistency_through_inputs(task)
+        def self.validate_mismatched_annotated_input(task, error_handler)
             task.each_annotated_port do |task_port, task_frame|
                 next if !task_port.input? || !task_frame
-                task_port.each_frame_of_connected_ports do |other_port, other_frame|
-                    if other_frame != task_frame
-                        raise FrameSelectionConflict.new(
-                            task,
-                            task.model.find_frame_of_port(task_port),
-                            task_frame,
-                            other_frame)
-                    end
-                end
+
+                task_port.each_frame_of_connected_ports
+                         .find_all { |other_port, other_frame| other_frame != task_frame }
+                         .each do |port, frame|
+                             conflict = FrameSelectionConflict.new(
+                                task,
+                                task.model.find_frame_of_port(task_port),
+                                task_frame,
+                                frame
+                             )
+                             error_handler.register_resolution_failures_from_exception(
+                                [task], conflict
+                             )
+                         end
             end
+        end
+
+        def self.validate_mismatched_transform_input(task, error_handler)
             task.each_transform_port do |task_port, task_transform|
                 next if !task_port.input?
-                task_port.each_transform_of_connected_ports do |other_port, other_transform|
-                    if other_transform.from && task_transform.from && other_transform.from != task_transform.from
-                        task_local_name = task.model.find_transform_of_port(task_port).from
-                        raise FrameSelectionConflict.new(task, task_local_name,
-                                                         task_transform.from, other_transform.from)
-                    elsif other_transform.to && task_transform.to && other_transform.to != task_transform.to
-                        task_local_name = task.model.find_transform_of_port(task_port).to
-                        raise FrameSelectionConflict.new(task, task_local_name,
-                                                         task_transform.to, other_transform.to)
+
+                from_connections_mismatch = task_port
+                    .each_transform_of_connected_ports.to_a
+                    .find_all do |other_port, other_tf|
+                        other_tf.from && task_transform.from &&
```

Comment:
Interestingly, whether `task_transform.from` is `nil` or not is fixed in the loop.

I think it would make sense to create `validate_mismatched_from_frame_in_transform_input` and ``validate_mismatched_to_frame_in_transform_input`, and test for nil-ness outside, e.g.

```
inputs = task.each_transform_port.find_all { |task_port, | task_port.input? }
inputs_with_from = inputs.find_all { |_, transform| transform.from }
inputs_with_to = inputs.find_all { |_, transform| transform.to }
validate_mismatched_from_frame_in_transform_input(iinputs_from, error_handler)
validate_mismatched_to_frame_in_transform_input(inputs_to, error_handler)
```